gateway/http_client.go

```diff
@@ -121,14 +130,20 @@ func newDefaultHTTPClientWithDebugMetrics() *httpClientWithDebugMetrics {
 // Logs detailed metrics and debugging information on failure for debugging.
 //
 // Returns: response body, HTTP status code, error
-func (h *httpClientWithDebugMetrics) SendHTTPRelay(
+func (h *HTTPClientWithDebugMetrics) SendHTTPRelay(
 	ctx context.Context,
 	logger polylog.Logger,
 	endpointURL string,
 	method string,
 	relayRequestBz []byte,
 	headers map[string]string,
 ) ([]byte, int, error) {
+	// Acquire concurrency slot before proceeding
+	if !h.limiter.acquire(ctx) {
+		return nil, 0, fmt.Errorf("failed to acquire concurrency slot: context canceled")
```

Comment:
This needs a TODO: use a predefined error, so we can build a proper observation indicating request failure due to reaching max concurrency.